packages/dd-trace/src/profiling/profilers/wall.js

```diff
@@ -31,16 +119,57 @@ class NativeWallProfiler {
       intervalMicros: this._samplingIntervalMicros,
       durationMillis: this._flushIntervalMillis,
       sourceMapper: this._mapper,
-      customLabels: this._codeHotspotsEnabled,
+      withContexts: this._codeHotspotsEnabled,
       lineNumbers: false
     })
 
+    if (this._codeHotspotsEnabled) {
+      this._profilerState = this._pprof.time.getState()
+      this._currentContext = {}
+      this._pprof.time.setContext(this._currentContext)
+
+      beforeCh.subscribe(this._enter)
+      enterCh.subscribe(this._enter)
+    }
+
     this._started = true
   }
 
-  profile () {
+  _enter () {
     if (!this._started) return
-    return this._pprof.time.stop(true)
+
+    if (this._profilerState[kSampleCount] !== 0) {
+      this._profilerState[kSampleCount] = 0
```

Comment:
Instead of resetting here, I'd rather keep a last sample count and compare to it
```
const sampleCount = this._profilerState[kSampleCount]
if (sampleCount != this._lastSampleCount) {
  ...
}
this._lastSampleCount = sampleCount
```
This way JS code never writes to it, only reads it.

Reply:
Nice, I like that !

Reply:
Done